cmake/pcl_find_vtk.cmake

```diff
@@ -0,0 +1,136 @@
+#Search for VTK
+find_package(VTK NO_MODULE)
+
+if(NOT VTK_FOUND)
+    return()
+endif()
+
+#Exit if version is below
+if("${VTK_VERSION}" VERSION_LESS 6.2)
+  message(WARNING "The minimum required version of VTK is 6.2, but found ${VTK_VERSION}")
+  set(VTK_FOUND FALSE)
+  return()
+endif()
+
+#Change to use list Transform when cmake is >= 3.12
+if(NOT (${VTK_VERSION} VERSION_LESS 9.0))
+set(PCL_VTK_COMPONENTS
+  ChartsCore
+  CommonColor
+  CommonCore
+  CommonDataModel
+  CommonExecutionModel
+  CommonMath
+  CommonMisc
+  CommonTransforms
+  FiltersCore
+  FiltersExtraction
+  FiltersGeneral
+  FiltersGeometry
+  FiltersModeling
+  FiltersSources
+  ImagingCore
+  ImagingSources
+  InteractionStyle
+  InteractionWidgets
+  IOCore
+  IOGeometry
+  IOImage
+  IOLegacy
+  IOPLY
+  RenderingAnnotation
+  RenderingCore
+  RenderingContext2D
+  RenderingLOD
+  RenderingFreeType
+  RenderingOpenGL2
+  ViewsCore
+  ViewsContext2D
+)
+else()
+set(PCL_VTK_COMPONENTS
+  vtkChartsCore
+  vtkCommonCore
+  vtkCommonDataModel
+  vtkCommonExecutionModel
+  vtkFiltersCore
+  vtkFiltersExtraction
+  vtkFiltersModeling
+  vtkImagingCore
+  vtkImagingSources
+  vtkInteractionStyle
+  vtkInteractionWidgets
+  vtkIOCore
+  vtkIOGeometry
+  vtkIOImage
+  vtkIOLegacy
+  vtkIOPLY
+  vtkRenderingAnnotation
+  vtkRenderingLOD
+  vtkViewsContext2D
+)
+endif()
```

Comment:
Indentation of `set` is missing.

Is it really necessary to have two lists? I had started more than one moth ago same test (just didn't had time to continue and as this PR also exists)...

I had solved it by:
```
function(checkVTKComponents)
  cmake_parse_arguments(PARAM "" "MISSING_COMPONENTS" "COMPONENTS" ${ARGN})

  set(vtkMissingComponents)
  if (VTK_VERSION VERSION_GREATER "8.2")
    foreach(vtkComponent ${PARAM_COMPONENTS})
      if (NOT TARGET VTK::${vtkComponent})
        list(APPEND vtkMissingComponents ${vtkComponent})
      endif()
    endforeach()
  else()
    foreach(vtkComponent ${PARAM_COMPONENTS})
      set(vtkComponent "vtk${vtkComponent}")
      if (NOT TARGET ${vtkComponent})
        list(APPEND vtkMissingComponents ${vtkComponent})
      endif()
    endforeach()
  endif()
  set(${PARAM_MISSING_COMPONENTS} ${vtkMissingComponents} PARENT_SCOPE)
endfunction()

set(PCL_VTK_COMPONENTS
  ChartsCore
  CommonCore
  CommonDataModel
  CommonExecutionModel
  FiltersCore
  FiltersExtraction
  FiltersModeling
  ImagingCore
  ImagingSources
  InteractionStyle
  InteractionWidgets
  IOCore
  IOGeometry
  IOImage
  IOLegacy
  IOPLY
  RenderingAnnotation
  RenderingLOD
  ViewsContext2D
)

# Start with a generic call to find any VTK version we are supporting, so we retrieve
# the version of VTK. As the module names were changed from VTK 8.2 to 9.0, we don't
# search explicitly for modules. Furthermore we don't pass required minimum version 6.2
# to find_package because then it only accept versions with same major version.
find_package(VTK)
if(VTK_FOUND AND ("${VTK_VERSION}" VERSION_LESS 6.2))
  message(WARNING "The minimum required version of VTK is 6.2, but found ${VTK_VERSION}")
  set(VTK_FOUND FALSE)
  return()
endif()

# We still require vtkOpenGLExtensionManager and vtkOpenGLRenderWindow
# Before VTK 8.2 they were part of VTK compontent
# Since VTK 8.2 they were part of VTK compontent vtkOpenGLRenderWindow => "OpenGL2", vtkOpenGLExtensionManager => missing
if("${VTK_VERSION}" VERSION_LESS 8.2)

else()

endif()

# Check if requested modules are available
checkVTKComponents(COMPONENTS ${PCL_VTK_COMPONENTS} MISSING_COMPONENTS vtkMissingComponents)
message(STATUS "${vtkMissingComponents}")
if (vtkMissingComponents)
  set(VTK_FOUND FALSE)
  message(WARNING "Missing vtk modules: ${vtkMissingComponents}")
  return()
endif()
```

Reply:
I'll look into using your snippet. 

Originally it was using List transform, but that wasn't available. But an "ordinary" foreach could be used ofc.